network/src/ledger.rs

```diff
@@ -510,12 +513,17 @@ impl<N: Network, E: Environment> Ledger<N, E> {
                 // Attempt to add the unconfirmed block as the next block in the canonical chain.
                 false => match self.canon.add_next_block(&unconfirmed_block) {
                     Ok(()) => {
+                        // Allocation necessary if prometheus is enabled.
+                        let latest_block_height = self.canon.latest_block_height();
                         info!(
                             "Ledger successfully advanced to block {} ({})",
-                            self.canon.latest_block_height(),
+                            latest_block_height,
                             self.canon.latest_block_hash()
                         );
 
+                        #[cfg(any(feature = "test", feature = "prometheus"))]
+                        metrics::gauge!(metrics::blocks::HEIGHT, latest_block_height as f64);
```

Comment:
this point isn't bad, but I wonder if the end of the happy path in `LedgerState::add_next_block` (as soon as `LedgerState::latest_block` is updated) is not a more future-proof place to update this metric :thinking:

Reply:
hmm, on second thought, I can see how `snarkos-storage` is not something that needs to be concerned with metrics; let's leave it as is, then :+1: